File: articles/educational-scaffolding-and-the-design-of-learning-systems/python/run_all_scaffolding_workflows.py
#!/usr/bin/env python3
"""Run all Python workflows for Educational Scaffolding."""

from pathlib import Path
import subprocess
import sys

ROOT = Path(__file__).resolve().parents[1]

scripts = [
    ROOT / "python" / "educational_scaffolding_audit_engine.py",
]

for script in scripts:
    print(f"Running {script.name}")
    subprocess.run([sys.executable, str(script)], check=True)

print("All Python scaffolding workflows complete.")

File: articles/educational-scaffolding-and-the-design-of-learning-systems/r/run_all_scaffolding_workflows.R
# run_all_scaffolding_workflows.R
# Run all R workflows for Educational Scaffolding.

args <- commandArgs(trailingOnly = FALSE)
file_arg <- grep("^--file=", args, value = TRUE)

if (length(file_arg) > 0) {
  script_path <- normalizePath(sub("^--file=", "", file_arg[1]), mustWork = TRUE)
  article_root <- normalizePath(file.path(dirname(script_path), ".."), mustWork = TRUE)
} else {
  article_root <- getwd()
}

source(file.path(article_root, "r", "educational_scaffolding_learning_system_analysis.R"))

message("All R scaffolding workflows complete.")
